joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/utils/IncludeAutoDiscovery.scala

```diff
@@ -66,36 +75,77 @@ object IncludeAutoDiscovery {
       Set.empty
   }
 
+  private def discoverMSVCInstallPath(): Option[String] = {
+    ExternalCommand.run(VsWhereCommand, ".") match {
+      case Success(output) =>
+        output.headOption
+      case Failure(exception) =>
+        logger.warn(s"Unable to discover MSVC installation path.", exception)
+        None
+    }
+  }
+
+  private def extractMSVCIncludePaths(resolvedInstallationPath: String): Set[Path] = {
+    ExternalCommand.run(VcVarsCommand, resolvedInstallationPath, Map("VSCMD_DEBUG" -> "3")) match {
+      case Success(results) =>
+        val includesLine   = results.find(_.startsWith("INCLUDE="))
+        val includesString = includesLine.map(_.replaceFirst("INCLUDE=", ""))
+        val includes       = includesString.map(_.split(";").toSet.map(p => Paths.get(p.trim).toRealPath()))
```

Comment:
Include path order can sometimes matter, so I think we want to use a linked hash set here, not just a plain set.